### PR TITLE
Add very large text scale support for time picker

### LINKED ISSUE #5939: Time picker should respect accessibility text scale factor
We currently respect text scale factors within the non-accessibility range. We should also add support for accessibility sizes (called "Larger Accessibility Sizes" on iOS).

### PR DESCRIPTION
**Pull request by guidezpl**
## Description

This PR adds support for very large text scales (called "Larger Accessibility Sizes" on iOS). 

### Before

Simulator Screen Shot - iPhone 11 Pro Max - 2020-01-14 at 16 26 09
Simulator Screen Shot - iPhone 11 Pro Max - 2020-01-14 at 16 26 05

### After

Simulator Screen Shot - iPhone 11 Pro Max - 2020-01-14 at 16 24 33
Simulator Screen Shot - iPhone 11 Pro Max - 2020-01-14 at 16 24 37

## Related Issues

Closes #5939 

## Tests

I added a test which ensures that certain elements are not affected by text scale (i.e. they are already large, like the time display), while others grow to a reasonable size.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/material/time_picker.dart [hunk 1]
```diff
@@ -261,7 +263,7 @@ class _DayPeriodControl extends StatelessWidget {
               heightFactor: 1,
               child: Semantics(
                 selected: amSelected,
-                child: Text(materialLocalizations.anteMeridiemAbbreviation, style: amStyle),
+                child: Text(materialLocalizations.anteMeridiemAbbreviation, style: amStyle, textScaleFactor: buttonTextScaleFactor),
```
Comment (rami-a) [inline packages/flutter/lib/src/material/time_picker.dart]: Nit: Add a trailing comma and format each parameter to its own line

File: packages/flutter/lib/src/material/time_picker.dart [hunk 2]
```diff
@@ -284,7 +286,7 @@ class _DayPeriodControl extends StatelessWidget {
               heightFactor: 1,
               child: Semantics(
                 selected: !amSelected,
-                child: Text(materialLocalizations.postMeridiemAbbreviation, style: pmStyle),
+                child: Text(materialLocalizations.postMeridiemAbbreviation, style: pmStyle, textScaleFactor: buttonTextScaleFactor),
```
Comment (rami-a) [inline packages/flutter/lib/src/material/time_picker.dart]: Same nit about trailing comma

**Review by rami-a — APPROVED** [review]

**Comment by HansMuller** [comment]
CC @darrenaustin

**Review by darrenaustin — APPROVED** [review]
LGTM

**Pull request merged**